### PR TITLE
site: update site type `{ children: React.ReactNode }` => `React.PropsWithChildren`

### PR DESCRIPTION
**Pull request by li-jia-nan**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | site: update site type `{ children: React.ReactNode }` => `React.PropsWithChildren` |
| 🇨🇳 Chinese | site: update site type `{ children: React.ReactNode }` => `React.PropsWithChildren` |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: .dumi/components/SemanticPreview.tsx [hunk 1]
```diff
@@ -65,12 +65,11 @@ const useStyle = createStyles(({ token }, markPos: [number, number, number, numb
 }));
 
 export interface SemanticPreviewProps {
```
Comment (afc163) [inline .dumi/components/SemanticPreview.tsx:67]:
```suggestion
export interface SemanticPreviewProps extends PropsWithChildren {
```

这样是不是好点。

Comment (li-jia-nan) [reply .dumi/components/SemanticPreview.tsx:67]: 这样结果也是没问题，但是感觉违背了 `PropsWithChildren` 的初衷，`PropsWithChildren` 内部的实现是：

```ts
type PropsWithChildren<P = unknown> = P & { children?: ReactNode | undefined };
```

这里的泛型是专门为了让用户传递 props 的类型

Comment (afc163) [reply .dumi/components/SemanticPreview.tsx:67]: 这样会修改 SemanticPreviewProps 本身的定义，children 没了。

Comment (li-jia-nan) [reply .dumi/components/SemanticPreview.tsx]: 这里我改回去了

**Review by afc163 — APPROVED** [review]

**Pull request merged**